src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -12,13 +23,374 @@
+            pd.DataFrame or Styler: dataframe containing results if style is False
+            If true it returns a Styler object
+        """
+
+        formatted_results = {}
+        for m, r in self.data.items():
+            metric_result = {}
+            for k, v in r.items():
+                if isinstance(v, dict):
+                    for vk, vv in v.items():
+                        if vk == "period_start_day":
+                            vk = "period"
+                        metric_result[f"{k}_{vk}"] = vv
+                else:
+                    metric_result[k] = v
+            formatted_results[m] = metric_result
+
+        df = pd.DataFrame.from_dict(formatted_results, orient="index")
+        df["effect_size_base_period"] = df["effect_size_period"] - 7
+        df["rel_effect_size"] = df["effect_size_value"] / df["mean_value"]
+
+        return df[
+            [
+                "relative_effect_size",
+                "effect_size_value",
+                "mean_value",
+                "std_dev_value",
+                "sample_size_per_branch",
+                "population_percent_per_branch",
+                "effect_size_period",
+                "mean_period",
+                "std_dev_period",
+            ]
+        ]
+
+    def get_styled_dataframe(self) -> Styler:
+        """returns styled dataframe for results from
+        empirical_effect_size_sample_size_calc
+
+        Arguments:
+            style (bool): If true, return a cleaned up and formatted pandas Styler.
+            Otherwise return a dataframe
+        Returns:
+            Styler: styled dataframe for visualization
+        """
+        df = self.get_dataframe()
+        return self.style_empirical_sizing_result(df)
+
+
+class SampleSizeCurveResultHolder(ResultsHolder):
+    def __init__(self, *args, **kwargs):
+        """
+        Args:
+            metrics (dict, optional): _description_. Defaults to None.
+            params (dict, optional): _description_. Defaults to None.
+        """
+        super().__init__(*args, **kwargs)
+        # need to modify results so it matches the old format
+        self.raw_df = pd.concat(
+            [el.set_index("effect_size", append=True) for el in self.data.values()]
+        )
+        metrics = [el.name for el in self._metrics]
+        results_dict = {}
+        for el in metrics:
+            results_dict[el] = self.raw_df.loc[el, :].reset_index()
+        self.data = results_dict
+
+    def set_raw_data_stats(self, input_data: pd.DataFrame) -> pd.DataFrame:
+        outlier_percentile = self._params["outlier_percentile"] / 100
+        overall_stats = (
+            input_data[[m.name for m in self._metrics]]
+            .agg(
+                [
+                    "mean",
+                    "std",
+                    lambda d: d[d <= d.quantile(outlier_percentile)].mean(),
+                    lambda d: d[d <= d.quantile(outlier_percentile)].std(),
+                ]
+            )
```

Comment:
These stats are also computed in all the solver functions (but don't get returned). Showing these stats is not specific to the sample size curves tool. I wonder if this should be a separate function at the top level of the module, and then it could get called in the `_get_sample_size_col` functions. We could then return these stats in another attr of the ResultsHolder. That would also help avoid having the trimming logic replicated in multiple places.

Within the context of this PR I think it's fine to leave this here, but we could consider making this change in a future one.

@m-d-bowerman What do think about updating the solver functions in this way? I usually recompute these stats after the fact when sizing because I want to see them alongside the sizing results

Reply:
This is a good change but for fear of messing up metric functions like `z_or_t_ind_sample_size_calc` I'd prefer to leave it for another PR.  If you or Michael feel strongly otherwise we can talk about it.